cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -330,15 +331,30 @@ Status TransferInt(RecordReader* reader, MemoryPool* pool,
   auto values = reinterpret_cast<const ParquetCType*>(reader->values());
   auto out_ptr = reinterpret_cast<ArrowCType*>(data->mutable_data());
   std::copy(values, values + length, out_ptr);
+  int64_t null_count = 0;
+  std::vector<std::shared_ptr<Buffer>> buffers = {nullptr, std::move(data)};
   if (field->nullable()) {
-    *out = std::make_shared<ArrayType<ArrowType>>(field->type(), length, std::move(data),
-                                                  reader->ReleaseIsValid(),
-                                                  reader->null_count());
-  } else {
-    *out =
-        std::make_shared<ArrayType<ArrowType>>(field->type(), length, std::move(data),
-                                               /*null_bitmap=*/nullptr, /*null_count=*/0);
+    null_count = reader->null_count();
+    buffers[0] = reader->ReleaseIsValid();
+  }
+  auto array_data =
+      ::arrow::ArrayData::Make(field->type(), length, std::move(buffers), null_count);
+  array_data->statistics.null_count = null_count;
+  auto statistics = metadata->statistics().get();
+  if (statistics) {
+    if (statistics->HasDistinctCount()) {
+      array_data->statistics.distinct_count = statistics->distinct_count();
+    }
+    if (statistics->HasMinMax()) {
+      auto typed_statistics =
+          static_cast<::parquet::TypedStatistics<ParquetType>*>(statistics);
+      array_data->statistics.min_buffer =
+          static_cast<ArrowCType>(typed_statistics->min());
+      array_data->statistics.max_buffer =
+          static_cast<ArrowCType>(typed_statistics->max());
+    }
   }
+  *out = std::make_shared<ArrayType<ArrowType>>(std::move(array_data));
```

Comment:
IIRC, the current parquet reader will return a chunked array containing one or multiple arrays (may or may not from different row groups). Here the stats do not contain the exact min/max values because they are from the row group level.

Reply:
Thanks for the information. I didn't know it.

It seems that arrays in a chunked array correspond to column chunks in Parquet, right?
If so, `ColumnChunkMetaData` for each column chunk has statistics for the column chunk, right? Or does it have statistics for the row group of the column chunk not the column chunk itself?

(Does you mean that we can't associate `ColumnChunkMetaData::statistics()` information with `arrow::Array` because it has the statistics for row group not column chunk?)

Reply:
The column statistics in the `ColumnChunkMetaData` is for each column chunk in that row group. However, if I remember correctly, the parquet reader may return arrow (chunked) arrays spanning more than one row group in a single `RecordBatch` or `Table`. I'm not sure if we can represent correct statistics in this case.

cc @mapleFU to confirm.

Reply:
In an extreme case, users might read a parquet file containing two row groups in a single `arrow::Table`, we have to merge the column statistics, right?

Reply:
We don't need to merge. Because each `arrow::Array` in a `arrow::Table` can have statistics.
We may add `arrow::Table`/`arrow::RecordBatch`/row group level statistics later but it's out of scope of this proposal.

Reply:
Sounds good. Then this is not an issue if stats is per array.